api/pkg/transformer/types/table/table.go

```diff
@@ -293,6 +293,27 @@ func (t *Table) Columns() []*series.Series {
 	return columns
 }
 
+func (t *Table) ColumnsWithExcluding(excludingCols []string) []*series.Series {
+	columnNames := t.ColumnNames()
+	columns := make([]*series.Series, 0)
+	for _, colName := range columnNames {
+		isExcluded := false
+		for _, excludingCol := range excludingCols {
```

Comment:
nit: for large number of columns this will be slightly slow. It's preferable to create a set based on the `excludingCols` and use the set as lookup table.